[python-nextgen] Fix validation of list of enums

t0mk opened the pull request:
@wing328 @spacether @krjakbrjak 

This PR fixes validation of lists of enums in python-nextgen, fixes #14785 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).

- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

t0mk commented on modules/openapi-generator/src/main/resources/python-nextgen/model_generic.mustache:
```diff
@@ -47,8 +47,14 @@ class {{classname}}({{#parent}}{{{.}}}{{/parent}}{{^parent}}BaseModel{{/parent}}
-            raise ValueError("must validate the enum values ({{#allowableValues}}{{#enumVars}}{{{value}}}{{^-last}}, {{/-last}}{{/enumVars}}{{/allowableValues}})")
+        if type(v) is list:
+            for i in v:
+                if i not in ({{#allowableValues}}{{#enumVars}}{{{value}}}{{^-last}}, {{/-last}}{{/enumVars}}{{/allowableValues}}):
```
@spacether Thanks for your comments! It was a quick fix, I wasn't thinking too much about it, and I'm not too familiar with mustache yet. I'll try to do the conditional generation next week.

I want to do the PR well, but I'm a beginner in mustache. If you want, you can drop some hints for me here, for example, how should I detect if schema is a list, maybe with a link to mustache template code that's already doing conditional based on schema type. Feel free to mention any other things that I should pay attention to.

spacether replied:
Please read about how mustache templates work. Here is a template of ours that checks for isArray and not isArray
https://github.com/OpenAPITools/openapi-generator/blob/0204bf4ae284d936e4f7cb1ffad314c04d5df9db/modules/openapi-generator/src/main/resources/objc/NSManagedObject-header.mustache#L13

t0mk replied:
@spacether, Thanks for the hints. I tried to fix your comment. I used `{{#isArray}}` as you requested. What do you think?

t0mk commented:
test failures:

```
=================================== FAILURES ===================================
____________________ ModelTests.test_inline_enum_validator _____________________

self = <tests.test_model.ModelTests testMethod=test_inline_enum_validator>

    def test_inline_enum_validator(self):
        self.pet = petstore_api.Pet(name="test name", photo_urls=["string"])
        self.pet.id = 1
        try:
>           self.pet.status = "error"

tests/test_model.py:319: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   pydantic.error_wrappers.ValidationError: 1 validation error for Pet
E   status
E     must be on of enum values ('available', 'pending', 'sold') (type=value_error)

pydantic/main.py:384: ValidationError

During handling of the above exception, another exception occurred:

self = <tests.test_model.ModelTests testMethod=test_inline_enum_validator>

    def test_inline_enum_validator(self):
        self.pet = petstore_api.Pet(name="test name", photo_urls=["string"])
        self.pet.id = 1
        try:
            self.pet.status = "error"
            self.assertTrue(False) # this line shouldn't execute
        except ValueError as e:
>           self.assertTrue("must validate the enum values ('available', 'pending', 'sold')" in str(e))
E           AssertionError: False is not true

tests/test_model.py:322: AssertionError

```

I'm not sure what's happening there, but I'd like to try. Can I run the tests locally somehow?

wing328 commented:
> I'm not sure what's happening there, but I'd like to try. Can I run the tests locally somehow?

yes, please refer to https://github.com/OpenAPITools/openapi-generator/wiki/Integration-Tests#how-to-add-integration-tests-for-new-petstore-samples. There are steps to run the petstore server locally.

t0mk commented:
@wing328 @spacether I fixed error check in tests, fixed typos, used `isArray`, and rebased to current master (16.03.2023). There's still some fail in the CircleCI test. 

I believe this is a legit bugfix, but I don't think I can do this any better, I simply don't have the time to dive into this project.

Feel free to take over.

wing328 commented:
@t0mk thanks for the PR. We'll take it from here.

wing328 commented:
CircleCI tests passed via https://app.circleci.com/pipelines/github/OpenAPITools/openapi-generator/22604/workflows/6d943ec4-4752-403d-8235-24d2d912a636
